src/platform/packages/private/kbn-scout-reporting/src/reporting/jest/reporter.ts

```diff
@@ -0,0 +1,256 @@
+        if (parsed.stack_trace) {
+          parsedStackTraces.push(parsed.stack_trace);
+        }
+      });
+
+    this.report.logEvent({
+      ...datasources.environmentMetadata,
+      reporter: {
+        name: this.name,
+        type: 'jest',
+      },
+      test_run: this.baseTestRunInfo,
+      suite: {
+        title: suiteTitle || 'unknown',
+        type: test.result.ancestorTitles.length <= 1 ? 'root' : 'suite',
+      },
+      test: {
+        id: getTestIDForTitle(test.result.fullName),
+        title: test.result.title,
+        tags: [],
+        file: this.getScoutFileInfoForPath(path.relative(REPO_ROOT, test.filePath)),
+        status: test.result.status === 'pending' ? 'skipped' : test.result.status,
+        duration: test.result.duration || 0,
+      },
+      event: {
+        action: ScoutReportEventAction.TEST_END,
+        error: {
+          message:
+            parsedErrorMessages.length > 0
+              ? parsedErrorMessages.join('\n--- NEXT ERROR ---\n')
+              : undefined,
+          stack_trace:
+            parsedStackTraces.length > 0
+              ? parsedStackTraces.join('\n--- NEXT STACK TRACE ---\n')
+              : undefined,
+        },
+      },
+    });
+  }
+
+  async onRunComplete(testContexts: Set<TestContext>, results: AggregatedResult): Promise<void> {
+    /**
+     * Test execution ended
+     */
+    // Load Scout metadata from Jest globals if a test context is found
+    const testContext = Array.from(testContexts.values()).find((context) => context !== undefined);
+
+    if (testContext !== undefined) {
+      this.loadMetadataFromJestGlobals(testContext.config.globals);
+    }
+
+    // Log "run start" event
+    this.report.logEvent({
+      ...datasources.environmentMetadata,
+      '@timestamp': new Date(results.startTime),
+      reporter: {
+        name: this.name,
+        type: 'jest',
+      },
+      test_run: this.baseTestRunInfo,
+      event: {
+        action: ScoutReportEventAction.RUN_BEGIN,
+      },
+    });
+
+    // Turn test results into events in bulk
+    results.testResults.forEach((suite) => {
+      suite.testResults.forEach((testResult) => {
+        this.logTestResult({ result: testResult, filePath: suite.testFilePath });
+      });
+    });
+
+    // Log "run end" event
+    this.report.logEvent({
+      ...datasources.environmentMetadata,
+      reporter: {
+        name: this.name,
+        type: 'jest',
+      },
+      test_run: {
+        ...this.baseTestRunInfo,
+        status: results.numFailedTests === 0 ? 'passed' : 'failed',
+        duration: Date.now() - results.startTime || 0,
+      },
+      event: {
+        action: ScoutReportEventAction.RUN_END,
+      },
+    });
+
+    // Save & conclude the report
+    try {
+      this.report.save(this.reportRootPath);
+      await uploadScoutReportEvents(this.report.eventLogPath, this.scoutLog);
+    } catch (e) {
+      // Log the error but don't propagate it
+      this.scoutLog.error(e);
```

Comment:
If we set the log level to `info` at the top, will this log ever be logged? I wonder if you got the chance to see this line actually logged.

From my understanding (correct me if I'm wrong), If the log level is info, it means only info-level logs and lower-severity logs (such as `success` and `warning`) will be logged.

Reply:
Any level with a value higher or equal in the [LogLevel enum](https://github.com/elastic/kibana/blob/54d9cf45cb0f3615b05414059e291d30d1c4a52d/src/platform/packages/shared/kbn-apm-synthtrace/src/lib/utils/create_logger.ts#L13-L18) will be sent to the log writer.

So, if you set the log level to `info`, then:
- `info`, `warn`, `error` will be logged
- `debug` and `verbose` won't

Reply:
Great info - thanks!